pynopegl-utils/pynopegl_utils/tests/cmp.py

```diff
@@ -96,22 +96,17 @@ def __init__(
         self._hud_export_filename = None
 
     def render_frames(self) -> Generator[Tuple[int, int, bytearray], None, None]:
-        cfg = ngl.SceneCfg(samples=self._samples, clear_color=self._clear_color)
-
         backend = os.environ.get("BACKEND")
-        if backend:
-            cfg.backend = get_backend(backend)
 
         width, height = self._width, self._height
-
         capture_buffer = bytearray(width * height * 4)
         ctx_cfg = ngl.Config(
             offscreen=True,
             width=width,
             height=height,
-            backend=cfg.backend if backend else ngl.Backend.AUTO,
-            samples=cfg.samples,
-            clear_color=cfg.clear_color,
+            backend=get_backend(backend) if backend else ngl.Backend.AUTO,
```

Comment:
I think this does not belong to the commit.

Reply:
I don't have `cfg` available anymore, so if I want to create an intermediate commit, I need to do something like that:
```python
         backend = os.environ.get("BACKEND")
         if backend:
-            cfg.backend = get_backend(backend)
+            backend = get_backend(backend)
```
But that introduces a mismatch of type (could be `str`, `None` or `Backend`), which would then be rechecked again with `backend=backend if backend else ngl.Backend.AUTO` which is not a good intermediate state I feel because it's pretty confusing what `backend` can be.